app/utils/goRoom.js

```diff
@@ -8,17 +8,36 @@ const navigate = ({ item, isMasterDetail, ...props }) => {
 		navigationMethod = Navigation.replace;
 	}
 
-	navigationMethod('RoomView', {
-		rid: item.rid,
-		name: RocketChat.getRoomTitle(item),
-		t: item.t,
-		prid: item.prid,
-		room: item,
-		search: item.search,
-		visitor: item.visitor,
-		roomUserId: RocketChat.getUidDirectMessage(item),
-		...props
-	});
+	if (item.isTeamChannel) {
```

Comment:
This is changing an abstraction that makes navigating to room to work both on mobile and tablet.
If you check other places, those extra steps are done before calling `goRoom`.